### PR TITLE
Cut changelog for v2.14

### PR DESCRIPTION
**Pull request by narqo**
#### What this PR does

This PR cuts the changelog for mimir v2.14, moving everything that isn't part of the r309 (~~or what we don't consider feature-ready~~) to the new unreleased section.

#### Which issue(s) this PR fixes or relates to

Relates to #9382

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,60 @@
...
+  * `cortex_alertmanager_state_replication_failed_total`
+  * `cortex_alertmanager_alerts`
+  * `cortex_alertmanager_silences`
+* [FEATURE] Querier: add experimental streaming PromQL engine, enabled with `-querier.query-engine=mimir`. #8422 #8430 #8454 #8455 #8360 #8490 #8508 #8577 #8660 #8671 #8677 #8747 #8850 #8872 #8838 #8911 #8909 #8923 #8924 #8925 #8932 #8933 #8934 #8962 #8986 #8993 #8995 #9008 #9017 #9018 #9019 #9120 #9121 #9136 #9139 #9140 #9145 #9191 #9192 #9194 #9196 #9201 #9212 #9225 #9260 #9272 #9277 #9278 #9280 #9281 #9342 #9343 #9367 #9368 #9371 #9398 #9399 #9403 #9417 #9418 #9419 #9420
+* [FEATURE] Experimental Kafka-based ingest storage. #6888 #6894 #6929 #6940 #6951 #6974 #6982 #7029 #7030 #7091 #7142 #7147 #7148 #7153 #7160 #7193 #7349 #7376 #7388 #7391 #7393 #7394 #7402 #7404 #7423 #7424 #7437 #7486 #7503 #7508 #7540 #7621 #7682 #7685 #7694 #7695 #7696 #7697 #7701 #7733 #7734 #7741 #7752 #7838 #7851 #7871 #7877 #7880 #7882 #7887 #7891 #7925 #7955 #7967 #8031 #8063 #8077 #8088 #8135 #8176 #8184 #8194 #8216 #8217 #8222 #8233 #8503 #8542 #8579 #8657 #8686 #8688 #8703 #8706 #8708 #8738 #8750 #8778 #8808 #8809 #8841 #8842 #8845 #8853 #8886 #8988
```
Comment (narqo) [inline CHANGELOG.md:17]: I'm moving both MQE and ingest-storage out of the 2.14. As discussed in the #9406 we are still working on those features.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,60 @@
...
+* [FEATURE] Query-frontend: added experimental configuration options `query-frontend.cache-errors` and `query-frontend.results-cache-ttl-for-errors` to allow non-transient responses to be cached. When set to `true` error responses from hitting limits or bad data are cached for a short TTL. #9028
+* [FEATURE] gRPC: Support S2 compression. #9322
+  * `-alertmanager.alertmanager-client.grpc-compression=s2`
+  * `-ingester.client.grpc-compression=s2`
+  * `-querier.frontend-client.grpc-compression=s2`
+  * `-querier.scheduler-client.grpc-compression=s2`
+  * `-query-frontend.grpc-client-config.grpc-compression=s2`
+  * `-query-scheduler.grpc-client-config.grpc-compression=s2`
+  * `-ruler.client.grpc-compression=s2`
+  * `-ruler.query-frontend.grpc-client-config.grpc-compression=s2`
```
Comment (narqo) [inline CHANGELOG.md:36]: I'm fine if we want to backport S2 compression into the release. Moved out for now.

File: CHANGELOG.md (hunk 1 above)
Comment (pstibrany) [reply CHANGELOG.md:17]: We don't do that. We keep PRs that were merged in the release section, even if we are not announcing the feature in release notes or have documentation for it.

Comment (pstibrany) [reply CHANGELOG.md:17]: For example "Querier: add experimental streaming PromQL engine," is already in 2.13 as well.

Comment (narqo) [reply CHANGELOG.md]: Ok, fair. I've put them back and split the mentioned MQE by PRs that are and aren't in the release.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,50 @@
...
+* [CHANGE] Alertmanager: the following metrics are not exported for a given `user` when the metric value is zero: #9359
+  * `cortex_alertmanager_alerts_received_total`
+  * `cortex_alertmanager_alerts_invalid_total`
+  * `cortex_alertmanager_partial_state_merges_total`
+  * `cortex_alertmanager_partial_state_merges_failed_total`
+  * `cortex_alertmanager_state_replication_total`
+  * `cortex_alertmanager_state_replication_failed_total`
+  * `cortex_alertmanager_alerts`
+  * `cortex_alertmanager_silences`
+* [FEATURE] Querier: add experimental streaming PromQL engine, enabled with `-querier.query-engine=mimir`. #9367 #9368 #9398 #9399 #9403 #9417 #9418 #9419 #9420
```
Comment (pracucci) [inline CHANGELOG.md:16]: All PRs that are included in v2.14.0-rc.0 should be mentioned in that CHANGELOG, even if the feature is experimental or incomplete. If a feature is incomplete, just mention it.

Comment (narqo) [reply CHANGELOG.md:16]: Do you mean to copy all PRs to keep them in both 2.14-rc.0 and unreleased? Right now, I've split the list of PRs into those that part of `2.14-rc.0` and those are new. That's how it was for this feature for 2.13 also

Comment (pstibrany) [reply CHANGELOG.md:16]: Those that are new (after 2.14-rc.0) are in `unreleased` section. I can confirm that #9367 and #3968 are only on `main` and not on `r309` (basis for 2.14 release).

**Review by pstibrany — APPROVED** [review]
Thank you.

**Pull request merged**